data/lib/src/network/config/endpoint.dart

```diff
@@ -66,4 +69,8 @@ extension ServicePathExtension on ServicePath {
   ServicePath append(ServicePath other) {
     return ServicePath(path + other.path);
   }
+
+  String generateSharingAutoCompletePath(AutoCompletePattern autoCompletePattern) {
+    return generateEndpointPath() + '/${autoCompletePattern.value}' + Endpoint.typeSharingAutoComplete;
```

Comment:
please find the way to generic add pathParameter

Reply:
Actually I did 
`String generateEndpointPathWithParameter(String parameter) {
    return '${Endpoint.rootPath}${path}/${parameter}';
  }
`
on Logout PR maybe we need an other for query parameters ?

Reply:
I created another method please check.